flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataDeserializationSchema.java

```diff
@@ -68,12 +69,16 @@
     /** Timestamp format specification which is used to parse timestamp. */
     private final TimestampFormat timestampFormat;
 
+    /** Flag indicating whether to allow unescaped control chars in the data. */
+    private final boolean allowUnescapedControlChars;
+
     public JsonRowDataDeserializationSchema(
             RowType rowType,
             TypeInformation<RowData> resultTypeInfo,
             boolean failOnMissingField,
             boolean ignoreParseErrors,
-            TimestampFormat timestampFormat) {
+            TimestampFormat timestampFormat,
+            boolean allowUnescapedControlChars) {
```

Comment:
The parameters are super large. Could you refactor it into builder pattern? Just like `CsvRowDataDeserializationSchema`. This can reduce the effort to add a new parameter in the future.